index.js

```diff
@@ -27,7 +27,6 @@ function start(func, port, cb, options) {
   const server = fastify({ logger: log });
 
   // All incoming requests get a Context object
-  server.decorateRequest('context');
   server.addHook('onRequest', (req, reply, done) => {
     req.context = new Context(req, reply);
```

Comment:
It's used right here! :)

And here: https://github.com/openshift-cloud-functions/faas-js-runtime/blob/master/lib/request-handler.js#L15-L20
And here: https://github.com/openshift-cloud-functions/faas-js-runtime/blob/master/lib/event-handler.js#L40-L46

The `decorateRequest` call ensures that there is a property called `context` on the request object within this scope, and line 32 here sets that to a new instance during the `onRequest` hook. The cloud event handler and the invoker both depend on this.

Reply:
Ah ok sorry, it looked to me that `onRequest` was adding the context property and the call to `decorateRequest` was not really doing anything apart from setting it to `undefined`.  And removing it work (test and manual run) so I thought it was not required 😞

Reply:
The two separate lines are how the Fastify docs recommend doing this. When you remove the call to `decorateRequest`, the `onRequest` hook still is invoked and the `context` property is still set. According to the docs, the `decorateRequest` call is optional, but allows Fastify to do a little performance optimization: https://www.fastify.io/docs/latest/Decorators/#usage-notes